sotodlib/site_pipeline/calibration/wiregrid.py

```diff
@@ -23,8 +29,56 @@
 
 logger = util.init_logger('wiregrid', 'wiregrid: ')
 
+def interpolate_hk(hk_data):
+    """
+    Simple function to get an interpolation function with a given house-keeping data loaded by so3g.hk.load_range.
+
+    Parameters
+    ----------
+        hk_data : data
+            house-keeping data as the output of so3g.hk.load_range
+
+    Returns
+    -------
+        interp_func : function
+            a function to interpolate the house-keeping data
+
+    """
+    interp_func = {}
+    for key, val in hk_data.items():
+        interp_func[key] = interp1d(hk_data[key][0], hk_data[key][1], bounds_error=False)
+    return interp_func
+
+
+# interp_func is the dictionary of functions to interpolate house-keeping data
+def cosamnple_hk(tod, interp_func, is_merge=False):
+    """
+    Simple function to co-sampling house-keeping data along the timestamps in tod. This function can be used for the general usage, but currently is specified to the wire grid calibration.
+
+    Parameters
+    ----------
+        tod : AxisManager
+        interp_func : function
+            output of the interpolate_hk
+        is_marge : bool (default, False)
+            if merge the result into tod or not
+
+    Returns
+    -------
+        hk_aman : AxisManager
+
+    """
+    hk_aman = core.AxisManager(tod.samps)
+    for key in interp_func.keys():
+        _cosamp_data = interp_func[key](tod.timestamps)
+        hk_aman.wrap(key, _cosamp_data, [(0, 'samps')])
+    if is_merge:
+        tod.wrap('hk_data', hk_aman)
+    return hk_aman
```

Comment:
Maybe it's fine, but the real data will be extrapolated by the standard cosampling methods. We don't want to extend the limit switch data because it can change the ON/OFF status. I added some comments just in case. In future pull request, they can be replaced by the common one though.